src/execution/dispatcher.js

```diff
@@ -23,33 +58,38 @@ export class Dispatcher {
     return Promise.resolve(data);
   }
 
+  hasPatches() {
+    return this._patches.length !== 0;
+  }
+
   add(
     label: string,
     path: Path | void,
-    fn: () => PromiseOrValue<mixed>,
+    fn: () => PromiseOrValue<ObjMap<mixed> | mixed>,
     errors: Array<GraphQLError>,
   ) {
     this._patches.push(
-      this.execute(fn).then((data) => ({
-        value: {
+      this.execute(fn).then((data) => {
+        const value: $Shape<ExecutionPatchResult> = {
```

Comment:
Can we add the `hasNext` here that's always `true` to remove the need for `$Shape`?

Reply:
it seems like its the errors spread that is confusing flow, not the missing `hasNext`